src/hooks/useLaunchableMarkets.ts

```diff
@@ -0,0 +1,93 @@
+import { useMemo } from 'react';
+
+import { useQuery } from '@tanstack/react-query';
+import { shallowEqual } from 'react-redux';
+
+import { useAppSelector } from '@/state/appTypes';
+import { getMarketIds } from '@/state/perpetualsSelectors';
+
+import { useSelectedNetwork } from './useSelectedNetwork';
+
+type MarketMapTicker = {
+  currency_pair: {
+    Base: string;
+    Quote: string;
+  };
+  decimals: string;
+  min_provider_count: string;
+  enabled: boolean;
+  metadata_JSON?: string;
+};
+
+type MarketMapProviderConfigs = {
+  name: string;
+  off_chain_ticker: string;
+  normalize_by_pair?: {
+    Base: string;
+    Quote: string;
+  };
+  invert: boolean;
+  metadata_JSON: string;
+};
+
+export type MarketMapResponse = {
+  chain_id: string;
+  last_updated: string;
+  market_map: {
+    markets: Record<
+      string,
+      {
+        ticker: MarketMapTicker;
+        provider_configs: MarketMapProviderConfigs[];
+      }
+    >;
+  };
+};
+
+export type HydratedMarketMap = MarketMapResponse['market_map']['markets'][string] & { id: string };
+
+export const useMarketMap = () => {
+  const { selectedNetwork } = useSelectedNetwork();
+
+  const launchableMarkets = useQuery({
+    enabled: selectedNetwork === 'dydxprotocol-staging',
+    queryKey: ['launchableMarkets', selectedNetwork],
+    queryFn: async () => {
+      return fetch('https://validator.v4staging.dydx.exchange:1317/slinky/marketmap/v1/marketmap')
```

Comment:
do we have a plan for not hardcoding this? I was wondering if we should @JsExport the V4StateManagerConfigs object from abacus which handles a lot of this

Reply:
Yeah the plan is to add the fetch to v4-client-js and expose from our useDydxClient hook.